wiki/DEFINITION_OF_DONE_CRITERIA.md

```diff
@@ -0,0 +1,95 @@
+- **Documentation:** User-facing documentation should be complete and descriptive. Developer-facing documentation should be used for constructs which are complex and/or not immediately obvious. 
+
+### C++
+
+#### Design
+
+- Existing prims are used wherever possible
+- Array inputs and outputs to algorithms are accepted on device
+- New prims created wherever there is potential for reuse across different algorithms or prims
+- User-facing API is stateless and follows the plain-old data (POD) design paradigm
+- Public API contains a C-Wrapper around the stateless API
+- Public API contains an sklearn-like stateful wrapper around the stateless API
+
+#### Testing
+
+- Prims: GTests with different inputs
+- Algorithms: End-to-end GTests with different inputs and different datasets
+
+#### Documentation
+
+- Complete and comprehensive Doxygen strings explaining the public API, restrictions, and gotchas. Any array parameters should also note whether the underlying memory is host or device. 
+
+
+### Python
+
+#### Design
+
+- Python class is as "near drop-in replacement" for sklearn (or relevant industry standard) API as possible
+- Python class is pickleable
+- APIs use `input_to_dev_array` to accept flexible inputs and check their datatypes
+
+#### Testing 
+
+- Pytests for wrapper functionality against sklearn using relevant skleaern datasets
+- Stress tests against reasonable inputs (e.g short-wide, tall-narrow)
+- Pytests for pickle capability
+- Pytests to evaluate correctness against sklearn on a variety of datasets
+- Add algorithm to benchmarks packages
+
+#### Documentation
+
+- Complete and comprehensive Pydoc strings explaining public API, restrictions, and gotchas
+- Docstrings include references to the underlying algorithm (e.g the paper or sklearn algo being implemented or a description of the algo used if nonstandard).
+
+
+## Algorithm Status
+
+Sometimes it is possible that only some of these requirements are satisfied before an algorithm gets released. We assign a status of `bronze`, `silver`, or `gold` to denote the amount of work necessary to provide the most robust, performant, and maintainable algorithm possible.
+
+### Bronze: 
+- Known design flaws affecting sustainability and/or performance require a significant effort / redesign
+- C++ checklist is complete but Python API has not been built
+- Python API exists without C++ API when a C++ implementation would provide a significant performance boost
+- There are known bugs that affect memory / accuracy on some inputs
+- Coverage of automated tests in C++ or Python is minimal and could be significantly improved
+
+
+### Silver:
+- Known design flaws affecting sustainability and/or performance are fairly low in complexity (can be fixed quickly and without too much effort)
+- Automated tests in C++ or Python are not as extensive as they could be, but they do test general functionality
```

Comment:
```suggestion
- Automated tests in C++ or Python are not as extensive as they could be regarding different datasets / inputs, but they have complete code coverage
```

Reply:
Using `have complete code coverage` seems to counter the `not extensive as they could be`. If the tests have complete code coverage, I'd say they are done. Our definitions of `complete` and `code coverage` might be different here.